hbw/app/javascript/packs/hbw/components/form.js.jsx

```diff
@@ -216,9 +216,44 @@ modulejs.define('HBWForm', ['React', 'jQuery', 'HBWError', 'HBWFormDatetime',
         return result;
       }, {});
 
-      this.setState({ formValues: variables });
+      this.setState({ formValues: this.updateSelectVariables(variables) });
     };
 
+    updateSelectVariables = (variables) => {
+      const updatingVariables = { ...variables };
+      const flattenFields = this.getFlattenFields(this.props.form.fields);
+      const fieldNamesforUpdate = this.getFilteredFieldNames(flattenFields);
+
+      fieldNamesforUpdate.forEach(((fieldName) => {
+        const updatingField = flattenFields.find(field => field.name === fieldName);
+        let value = updatingField.choices[0];
+
+        if (Array.isArray(value)) {
+          value = value[0];
+        }
+
+        updatingVariables[fieldName] = value;
+      }));
+
+      return updatingVariables;
+    }
```

Comment:
I think it's better to do this inside general `reduce` not after that.

Also, let's remove the variable `value` mutation.